app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -0,0 +1,361 @@
+      if (suggestedMemoryFactor < UnifiedMemoryHeuristic.SPARK_MEMORY_FRACTION_THRESHOLD()) {
+        suggestedMemoryFactor = UnifiedMemoryHeuristic.SPARK_MEMORY_FRACTION_THRESHOLD();
+      }
+      suggestedParameters.put(SparkConfigurationConstants.SPARK_MEMORY_FRACTION_KEY, suggestedMemoryFactor);
+      suggestedParameters.put(SparkConfigurationConstants.SPARK_DRIVER_MEMORY_KEY, (double) suggestedDriverMemory
+          / FileUtils.ONE_MB);
+      logger.info("Following are the suggestions for spark parameters for app id : " + appResult.flowExecId);
+      logger.info("suggestedExecutorMemory " + suggestedExecutorMemory);
+      logger.info("suggestedCore " + suggestedCore);
+      logger.info("suggestedMemoryFactor " + suggestedMemoryFactor);
+      logger.info("suggestedDriverMemory " + suggestedDriverMemory);
+    } catch (Exception e) {
+      logger.error("Error in generating parameters ", e);
+    }
+    return suggestedParameters;
+  }
+
+  private Long getMaxPeakJVMUsedMemoryPerCore() {
+    return maxPeakJVMUsedMemory / lastRunExecutorCore;
+  }
+
+  private Long getMaxPeakUnifiedMemoryPerCore() {
+    return maxPeakUnifiedMemory / lastRunExecutorCore;
+  }
+
+  /**
+   * Suggest executor memory and core We calculate per core peak JVM used memory. Then we start from
+   * max executor core and compute memory requirement for max executor core If this is below
+   * threshold of max executor core then good, otherwise reduce the core and check memory
+   * requirement.
+   */
+  private void suggestExecutorMemoryCore() {
+    boolean validSuggestion = false;
+    long maxPeakJVMUsedMemoryPerCore = getMaxPeakJVMUsedMemoryPerCore();
+    for (int core = MAX_EXECUTOR_CORE; core > 0; core--) {
+      long currSuggestedMemory = suggestExecutorMemory(maxPeakJVMUsedMemoryPerCore, core);
+      if (currSuggestedMemory < MAX_EXECUTOR_MEMORY || core == 1) {
+        suggestedExecutorMemory = currSuggestedMemory;
+        suggestedCore = core;
+        validSuggestion = true;
+        break;
+      }
+    }
+    if (!validSuggestion) {
+      suggestedExecutorMemory = lastRunExecutorMemory;
+      suggestedCore = lastRunExecutorCore;
+    }
+    if (getMemoryIncreaseForGC() != 0) {
+      suggestedExecutorMemory = suggestedExecutorMemory * (100 + getMemoryIncreaseForGC()) / 100;
+    }
+    if (suggestedExecutorMemory < MIN_EXECUTOR_MEMORY) {
+      suggestedExecutorMemory = MIN_EXECUTOR_MEMORY;
+    }
+    suggestedExecutorMemory = getRoundedExecutorMemory();
+  }
+
+  /**
+   * This method is to get the rounded executor memory. We don't want to round executor memory to
+   * multiple of 1GB as container size is decided by executor memory + memoryOverhead. So round the
+   * executor memory + memoryOverhead and then distribute extra memory to executor memory
+   *
+   * @return rounded executor memory
+   */
+  private long getRoundedExecutorMemory() {
```

Comment:
@varunsaxena 
See below for @mkumar1984 response 
https://github.com/linkedin/dr-elephant/pull/430/commits/27c04114e9b219feb15d0bc08d6e88a56fe6a7f3#r223654515

Reply:
Can't see any comment when I click on the link. So would this be fixed in next PR?

Reply:
As per @mkumar1984 ,currently the code is same ,but it may change latter on , that's why its better to keep it separate

Reply:
If it's going to change in the very next PR then its fine. If it may change eventually, its better to combine the methods. These are private methods, they can be split into two anytime. 
Code duplication in general is not desirable.